### PR TITLE
Bump libyang, libnetconf2, sysrepo and netopeer2

### PR DESCRIPTION
**Pull request by mattiaswal**
For netopeer2 we need to patch away local-users, since if it is enabled netopeer2 requires all users that should be allowed to use netconf to also be configured in ietf-netconf-server.

Also add some upstream patches.

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by mattiaswal** [comment]
make legal-check: OK
tests on hw: PASS

**Comment by troglobit** [comment]
Looks good, how about a fresh new changelog to kick off the new release cycle?

File: src/confd/yang/confd.inc [hunk 1]
```diff
@@ -43,7 +43,6 @@ MODULES=(
 	"ietf-netconf@2013-09-29.yang -e writable-running -e candidate -e rollback-on-error -e validate -e startup -e url -e xpath -e confirmed-commit"
 	"ietf-truststore@2023-12-28.yang -e central-truststore-supported -e certificates"
 	"ietf-keystore@2023-12-28.yang -e central-keystore-supported -e inline-definitions-supported -e asymmetric-keys -e symmetric-keys"
-	"ietf-ssh-server@2023-12-28.yang -e local-user-auth-password -e local-user-auth-publickey"
 	"ietf-tls-server@2023-12-28.yang -e server-ident-raw-public-key -e server-ident-x509-cert"
```
Comment (troglobit) [inline src/confd/yang/confd.inc:46]: This is the only part I'm slightly worried about, shouldn't it match the changes to netopeer2/scripts/common.sh?

**Review by troglobit — APPROVED** [review]

File: src/confd/yang/confd.inc (hunk 1 above)
Comment (mattiaswal) [reply src/confd/yang/confd.inc:46]: Could do, but not required. I think we should remove all that we do not need to change from that file. If we should add extra features it should of course remain.

Comment (troglobit) [reply src/confd/yang/confd.inc:46]: Yeah, valid point!  

Nice work on this, btw!

**Pull request merged**